9.2.1 arm gcc does not compile rusefi with lua change

```
Compiling init_lambda.cpp
Compiling init_map.cpp
Compiling init_flex.cpp
Compiling init_vbatt.cpp
Compiling init_baro.cpp
Compiling init_fuel_level.cpp
Compiling rusefi.cpp
Compiling main.cpp
rusefi.cpp: In function 'bool validateConfig()':
rusefi.cpp:165:87: warning: comparison of integer expressions of different signedness: 'cylinders_count_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
  165 |  if (CONFIG(specs.cylindersCount) > minI(INJECTION_PIN_COUNT, IGNITION_PIN_COUNT)) {
      |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^             
Linking build/rusefi.elf
c:/program files (x86)/gnu tools arm embedded/9 2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld.exe: c:/program files (x86)/gnu tools arm embedded/9 2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard\libg_nano.a(lib_a-abort.o): in function `abort':
abort.c:(.text.abort+0xa): undefined reference to `_exit'
collect2.exe: error: ld returned 1 exit status
make: *** [ChibiOS/os/common/startup/ARMCMx/compilers/GCC/mk/rules.mk:217: build/rusefi.elf] Error 1
mv: cannot stat 'build/rusefi.elf': No such file or directory
mv: cannot stat 'build/rusefi.hex': No such file or directory
cp: cannot stat 'build/rusefi.bin': No such file or directory
Debug compilation results 2/2
TIMESTAMP 05/09/21 16:42:38.91
total 2888
```

![image](https://user-images.githubusercontent.com/48498823/117586520-ae68c300-b0e6-11eb-9d41-ccee3fd19fdd.png)

i can bump gcc to 2021

do we need some sort of version check in Makefile?

--- a/firmware/main.cpp
+++ b/firmware/main.cpp
@@ -11,6 +11,7 @@
 #include "os_access.h"
 #include "rusefi.h"
 #include "mpu_util.h"
+#include "version_check.h"
 
 int main(void) {
 	/*

--- a/firmware/version_check.h
+++ b/firmware/version_check.h
@@ -1,0 +1,13 @@
+// This file asserts that the compiler is appropriate for rusEFI use.
+
+#pragma once
+
+// non-MCU builds are significantly more tolerant
+#if EFI_PROD_CODE
+
+static constexpr auto gccVersion = (__GNUC__ * 100) + (__GNUC_MINOR__ * 10) + ( __GNUC_PATCHLEVEL__ );
+
+// Firmware builds require at least GCC 9.3.1
+static_assert(gccVersion >= 931, "GCC compiler >= 9.3.1 required");
+
+#endif

